### PR TITLE
[stable/sugarcrm] adds chart to deploy SugarCRM

### PR DESCRIPTION
**Pull request by sameersbn**

### CODE REVIEW

**Review by unguiculus — CHANGES_REQUESTED** [review]
Please apply best practices for standard labels, especially the `app` label. 

https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/labels.md

**Comment by spesnova** [comment]
@sameersbn 

It looks this chart doesn't work with AWS ELB.

I tried to install this chart to a Kubernetes cluster (1.6+) on AWS. All resources were created correctly. I could see pods were ready status, an ELB was created. But ELB healthcheck result did not change to healthy.  I guess sugarcrm container's redirecting is the cause.

> If the load balancer receives any response other than "200 OK" within the response timeout period, the instance is considered unhealthy.

http://docs.aws.amazon.com/elasticloadbalancing/latest/classic/elb-healthchecks.html

So sugarcrm(apache) should not redirect traffic from ELB.

**Comment by unguiculus** [comment]
Marking as stale. Please update within the next week.

**Comment by unguiculus** [comment]
Closing as stale. Feel free to re-open if/when you decide to work on this again.

**Comment by sameersbn** [comment]
updated chart with suggested changes.

File: stable/sugarcrm/templates/_helpers.tpl [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 24 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
```
Comment (unguiculus) [inline stable/sugarcrm/templates/_helpers.tpl]: 24 -> 63

File: stable/sugarcrm/templates/_helpers.tpl [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+*/}}
+{{- define "fullname" -}}
+{{- $name := default .Chart.Name .Values.nameOverride -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 24 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
```
Comment (unguiculus) [inline stable/sugarcrm/templates/_helpers.tpl]: 24 -> 63

**Review by unguiculus — CHANGES_REQUESTED** [review]
The `app` label should be `{{ template "name" . }}`.

**Comment by sameersbn** [comment]
updated the pull-request with the required changes.

**Comment by sameersbn** [comment]
updated with the required changes.

File: stable/sugarcrm/templates/NOTES.txt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+  helm upgrade {{ .Release.Name }} \
+    --set sugarcrmHost=$APP_HOST,sugarcrmPassword=$APP_PASSWORD{{ if .Values.mariadb.mariadbRootPassword }},mariadb.mariadbRootPassword=$APP_DATABASE_PASSWORD{{ end }} stable/sugarcrm
+
+{{- else -}}
+1. Get the SugarCRM URL by running:
+
+{{- if eq .Values.serviceType "ClusterIP" }}
+
+  export POD_NAME=$(kubectl get pods --namespace {{ .Release.Namespace }} -l "app={{ template "fullname" . }}" -o jsonpath="{.items[0].metadata.name}")
```
Comment (unguiculus) [inline stable/sugarcrm/templates/NOTES.txt]: `"app={{ template "fullname" . }}"` -> `"app={{ template "name" . }},release={{ .Release.Name }}"`

File: stable/sugarcrm/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  name: {{ template "fullname" . }}
+  labels:
+    app: {{ template "name" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+spec:
+  template:
+    metadata:
+      labels:
```
Comment (unguiculus) [inline stable/sugarcrm/templates/deployment.yaml:14]: Only `app` and `release` label should be added here.

File: stable/sugarcrm/templates/svc.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  {{- end }}
+  ports:
+  - name: http
+    port: 80
+    targetPort: http
+  - name: https
+    port: 443
+    targetPort: https
+  selector:
+    app: {{ template "name" . }}
```
Comment (unguiculus) [inline stable/sugarcrm/templates/svc.yaml:25]: Add `release` label to selector.

File: stable/sugarcrm/templates/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+        app: {{ template "name" . }}
+        chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+        release: "{{ .Release.Name }}"
+        heritage: "{{ .Release.Service }}"
+    spec:
+      containers:
+      - name: {{ template "fullname" . }}
+        image: "{{ .Values.image }}"
+        imagePullPolicy: {{ default "" .Values.imagePullPolicy | quote }}
```
Comment (unguiculus) [inline stable/sugarcrm/templates/deployment.yaml]: No need to add `default "" `.

File: stable/sugarcrm/templates/deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+          value: "3306"
+        - name: MARIADB_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: {{ template "mariadb.fullname" . }}
+              key: mariadb-root-password
+        - name: SUGARCRM_HOST
+          value: {{ include "host" . | quote }}
+        - name: SUGARCRM_USERNAME
+          value: {{ default "" .Values.sugarcrmUsername | quote }}
```
Comment (unguiculus) [inline stable/sugarcrm/templates/deployment.yaml]: No need to add `default "" ` if you add an empty default to `values.yml`. Apply where applicable in the whole chart.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by sameersbn** [comment]
updated! thanks for the review,

File: stable/sugarcrm/templates/NOTES.txt [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+  helm upgrade {{ .Release.Name }} \
+    --set sugarcrmHost=$APP_HOST,sugarcrmPassword=$APP_PASSWORD{{ if .Values.mariadb.mariadbRootPassword }},mariadb.mariadbRootPassword=$APP_DATABASE_PASSWORD{{ end }} stable/sugarcrm
+
+{{- else -}}
+1. Get the SugarCRM URL by running:
+
+{{- if eq .Values.serviceType "ClusterIP" }}
+
+  export POD_NAME=$(kubectl get pods --namespace {{ .Release.Namespace }} -l "app={{ template "name" . }}" -o jsonpath="{.items[0].metadata.name}")
```
Comment (unguiculus) [inline stable/sugarcrm/templates/NOTES.txt]: `"app={{ template "name" . }}"` -> `"app={{ template "name" . }}",release="{{ .Release.Name }}"`

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/sugarcrm/templates/NOTES.txt (hunk 8 above)
Comment (sameersbn) [reply stable/sugarcrm/templates/NOTES.txt]: good catch!

**Comment by sameersbn** [comment]
updated! thanks for the in-depth review. great work!

**Review by unguiculus — APPROVED** [review]

**Pull request merged**